Hmm, this consistently fails to build for me:

```
2024/06/15 20:56:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/go-build3765734460/b001/data/actions_artifacts

[1] Test Check
OK

All done (checks: 1).
--- FAIL: TestAddRecursiveExclude (0.00s)
    --- FAIL: TestAddRecursiveExclude/File_inside_directory (0.00s)
        --- FAIL: TestAddRecursiveExclude/File_inside_directory/No_exclude (0.00s)
            dump_test.go:83:
                        Error Trace:    /build/source/cmd/dump_test.go:83
                        Error:          Not equal:
                                        expected: "deep/nested/folder/example"
                                        actual  : "deep/nested/folder/another-file"

                                        Diff:
                                        --- Expected
                                        +++ Actual
                                        @@ -1 +1 @@
                                        -deep/nested/folder/example
                                        +deep/nested/folder/another-file
                        Test:           TestAddRecursiveExclude/File_inside_directory/No_exclude
            dump_test.go:84:
                        Error Trace:    /build/source/cmd/dump_test.go:84
                        Error:          Not equal:
                                        expected: "deep/nested/folder/another-file"
                                        actual  : "deep/nested/folder/example"

                                        Diff:
                                        --- Expected
                                        +++ Actual
                                        @@ -1 +1 @@
                                        -deep/nested/folder/another-file
                                        +deep/nested/folder/example
                        Test:           TestAddRecursiveExclude/File_inside_directory/No_exclude

Forgejo: Protocol: format length error

Forgejo: Protocol: format length error

Forgejo: Protocol: write error

Forgejo: Protocol: write error

2024/06/15 20:56:11 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/TestMigratePackages255460776/001
2024/06/15 20:56:11 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/TestMigrateActionsArtifacts2597845411/001
2024/06/15 20:56:11 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/TestMigrateActionsArtifacts2597845411/002
2024/06/15 20:56:11 ...d/migrate_storage.go:174:func3() [W] ignored: actions artifact /notfound exists in the database but not in storage
FAIL
FAIL    code.gitea.io/gitea/cmd 1.012s
FAIL
error: builder for '/nix/store/l1cgsk6jmnjk7i8jr2d5ha1hpggdics4-forgejo-7.0.4.drv' failed with exit code 1;
       last 10 log lines:
       >
       > Forgejo: Protocol: write error
       >
       > 2024/06/15 20:56:11 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/TestMigratePackages255460776/001
       > 2024/06/15 20:56:11 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/TestMigrateActionsArtifacts2597845411/001
       > 2024/06/15 20:56:11 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/TestMigrateActionsArtifacts2597845411/002
       > 2024/06/15 20:56:11 ...d/migrate_storage.go:174:func3() [W] ignored: actions artifact /notfound exists in the database but not in storage
       > FAIL
       > FAIL    code.gitea.io/gitea/cmd 1.012s
       > FAIL
       For full logs, run 'nix log /nix/store/l1cgsk6jmnjk7i8jr2d5ha1hpggdics4-forgejo-7.0.4.drv'.
```

--- a/pkgs/by-name/fo/forgejo/package.nix
+++ b/pkgs/by-name/fo/forgejo/package.nix
@@ -2,6 +2,7 @@
 , brotli
 , buildGoModule
 , forgejo
+, fetchpatch
 , git
 , gzip
 , lib
@@ -55,11 +56,21 @@ buildGoModule rec {
 
   outputs = [ "out" "data" ];
 
-  nativeBuildInputs = [ makeWrapper ];
+  nativeBuildInputs = [
+    makeWrapper
+    git # checkPhase
+    openssh # checkPhase
+  ];
   buildInputs = lib.optional pamSupport pam;
 
   patches = [
     ./static-root-path.patch
+    # Fix TestAddRecursiveExclude go test.
+    # This will be part of the upcoming v7.0.5 release at which point this needs to be removed again.
+    (fetchpatch {
+      url = "https://codeberg.org/forgejo/forgejo/commit/f01dc4b271f54201e60a7c795d784813eac3f7ce.patch";
+      sha256  = "sha256-1lPLVHBad+Fmk+6EFXKuMnmDUl5BkEYJuMkTPfdrCgU=";
+    })
   ];
 
   postPatch = ''
@@ -79,6 +90,33 @@ buildGoModule rec {
   preConfigure = ''
     export ldflags+=" -X main.ForgejoVersion=$(GITEA_VERSION=${version} make show-version-api)"
   '';
+
+  preCheck = ''
+    # $HOME is required for ~/.ssh/authorized_keys and such
+    export HOME="$TMPDIR/home"
+
+    # expose and use the GO_TEST_PACKAGES var from the Makefile
+    # instead of manually copying over the entire list:
+    # https://codeberg.org/forgejo/forgejo/src/tag/v7.0.4/Makefile#L124
+    echo -e 'show-backend-tests:\n\t@echo ''${GO_TEST_PACKAGES}' >> Makefile
+    getGoDirs() {
+      make show-backend-tests
+    }
+  '';
+
+  checkFlags =
+    let
+      skippedTests = [
+        "Test_SSHParsePublicKey/dsa-1024/SSHKeygen" # dsa-1024 is deprecated in openssh and requires opting-in at compile time
+        "Test_calcFingerprint/dsa-1024/SSHKeygen" # dsa-1024 is deprecated in openssh and requires opting-in at compile time
+        "TestPamAuth" # we don't have PAM set up in the build sandbox
+        "TestPassword" # requires network: api.pwnedpasswords.com
+        "TestCaptcha" # requires network: hcaptcha.com
+        "TestDNSUpdate" # requires network: release.forgejo.org
+        "TestMigrateWhiteBlocklist" # requires network: gitlab.com (DNS)
+      ];
+    in
+    [ "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$" ];
 
   postInstall = ''
     mkdir $data

